src/storage/src/opts.rs

```diff
@@ -30,8 +31,6 @@ pub struct StorageOpts {
     /// Maximum shared buffer size, writes attempting to exceed the capacity will stall until there
     /// is enough space.
     pub shared_buffer_capacity_mb: usize,
-    /// State store url.
-    pub state_store: String,
     /// Remote directory for storing data and metadata objects.
     pub data_directory: String,
```

Comment:
`state_store` is removed but `data_directory` is still here? Should we move them together (or merge into one)? (mentioned in that [Notion doc](https://www.notion.so/risingwave-labs/System-Parameters-b9b9c00308bc4020a4208fa1764b24db))

Reply:
`StorageOpts` is intended for internal use by storage code, `state_store` is removed simply because it is not used.

But I indeed removed `state_store` from config. The reason is I added it in #7613 (in release 0.1.17), but I soon realized it is pointless since it will be removed in 0.1.18 (since system params don't appear in the config file), so I can save the trouble of maintaining backward compatibility by removing it right now 🤣.

> Should we move them together (or merge into one)?

This is a breaking change. Shall we change it now?